src/utils/staking/index.ts

```diff
@@ -43,3 +63,133 @@ export const buildStakingOutput = (
   }
   return psbtOutputs;
 };
+
+/**
+ * Get the staking term, it will ignore the `stakingTimeBlocks` and
+ * use the value from params if the min and max staking time blocks are the same
+ *
+ * @param {Params} params - The staking parameters.
+ * @param {number} term - The staking term.
+ * @returns {number} - The staking term.
+ */
+export const getStakingTerm = (params: Params, term: number) => {
+  // check if term is fixed
+  let termWithFixed;
+  if (params.minStakingTimeBlocks === params.maxStakingTimeBlocks) {
+    // if term is fixed, use the API value
+    termWithFixed = params.minStakingTimeBlocks;
+  } else {
+    // if term is not fixed, use the term from the input
+    termWithFixed = term;
+  }
+  return termWithFixed;
+};
+
+/**
+ * Validate the staking transaction input data.
+ *
+ * @param {number} stakingAmountSat - The staking amount in satoshis.
+ * @param {number} stakingTerm - The staking term in blocks.
+ * @param {Params} params - The staking parameters.
+ * @param {UTXO[]} inputUTXOs - The input UTXOs.
+ * @param {number} feeRate - The Bitcoin fee rate in sat/vbyte
+ * @throws {StakingError} - If the input data is invalid.
+ */
+export const validateStakingTxInputData = (
+  stakingAmountSat: number,
+  stakingTerm: number,
```

Comment:
it seems this functions is good place to validate that , `stakingTerm` should be at most 65536 i.e max uint16 number